apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx

```diff
@@ -1,41 +1,67 @@
 import { contributors } from '@fmc/data/content';
-import { getSortedChallengesByDifficulty } from '@fmc/data/utils';
+import { getSortedChallengesByDifficulty, getSortedChallengesByTitle } from '@fmc/data/utils';
 import styles from './challenge-grid.module.scss';
 import { AvatarGroup } from '../avatar/avatar';
 import { IChallenge } from '@fmc/data/types';
+import { useEffect, useState } from 'react';
+import assets from '@fmc/assets/images';
+import { Link } from 'react-router-dom';
 
 interface Props {
   challenges: Map<string, IChallenge>;
   linkPrefix: string;
+  links: {tech: string, imgSrc: string, active: boolean }[];
 }
 
-function ChallengeGrid({ challenges, linkPrefix }: Props) {
-  const sortedChallengesByDifficulty = getSortedChallengesByDifficulty(challenges);
+function ChallengeGrid({ challenges, linkPrefix, links }: Props) {
+  const [searchInput, setSearchInput] = useState<string>('');
+  const [sortedChallenges, setSortedChallenges] = useState(() => getSortedChallengesByDifficulty(challenges));
+
+  useEffect(() => {
+    if (searchInput) {
+      setSortedChallenges(() => getSortedChallengesByTitle(challenges, searchInput))
+    } else {
+      setSortedChallenges(() => getSortedChallengesByDifficulty(challenges));
+    }
+  }, [challenges, searchInput]);
 
   return (
-    <div className={styles.challengeGrid}>
-      {sortedChallengesByDifficulty.map((challenge) => (
-        <a
-          key={challenge.title}
-          className={`${styles.challengeCard} ${styles[challenge.difficulty]}`}
-          href={linkPrefix + challenge.link}
-        >
-          {challenge.isNew && <span className={styles.new}>New</span>}
-          <div>
-            <h3>{challenge.title}</h3>
+    <div className={styles.container}>
+      <div className={styles.filterOptionWrapper}>
+        <div className={styles.searchInputWrapper}>
+          <input type="text" name="searchTextInput" placeholder='Search challenge...' className={styles.searchInput} value={searchInput} onChange={(e) => setSearchInput(e.target.value.trim())} />
+          <img src={assets.searchIconSVG} alt="search challenges by title" width={15} height={15} className={styles.searchIcon} />
+        </div>
+        <div className={styles.filterByTechWrapper}>
+          {links.map((link) => <Link to={`/${link.tech}`} key={link.tech}>
+            <img src={link.imgSrc} width={35} height={35} className={link.active ? styles.activeTech : ''} alt={`filter by ${link.tech}`} />
+          </Link>)}
+        </div>
+      </div>
```

Comment:
can we move the above one as a separate component

Reply:
the whole filter row? or only filter by tech div? @sadanandpai

Reply:
whole

Reply:
I have created another pr, as this was already merged. Please review this [pr](https://github.com/sadanandpai/frontend-mini-challenges/pull/402). 🙂